src/cls/tabular/cls_tabular.cc

```diff
@@ -1517,7 +1504,24 @@ int exec_query_op(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
                                            meta.blob_size,
                                            errmsg,
                                            row_nums);
-                        // TODO: Add the output to result ds
+
+                        if (ret != 0) {
+                            CLS_ERR("ERROR: processSkyFb %s", errmsg.c_str());
+                            CLS_ERR("ERROR: TablesErrCodes::%d", ret);
+                            return -1;
+                        }
+
+                        std::shared_ptr<arrow::Buffer> buffer;
+                        convert_arrow_to_buffer(table, &buffer);
+                        createFbMeta(meta_builder,
+                                     SFT_ARROW,
+                                     (unsigned char *)buffer->ToString().c_str(),
+                                     buffer->size());
+
+                        if (op.index_read)
```

Comment:
I think you can pull this out and put below the cases.

Reply:
Done